pkg/3scale/amp/operator/apicast_reconciler.go

```diff
@@ -151,8 +157,30 @@ func (r *ApicastReconciler) apicastLogLevelEnvVarMutator(desired, existing *apps
 	return reconcilers.DeploymentConfigEnvVarReconciler(desired, existing, "APICAST_LOG_LEVEL")
 }
 
-func Apicast(apimanager *appsv1alpha1.APIManager) (*component.Apicast, error) {
-	optsProvider := NewApicastOptionsProvider(apimanager)
+func (r *ApicastReconciler) volumeMountsMutator(desired, existing *appsv1.DeploymentConfig) bool {
+	changed := false
+
+	if !reflect.DeepEqual(existing.Spec.Template.Spec.Containers[0].VolumeMounts, desired.Spec.Template.Spec.Containers[0].VolumeMounts) {
+		changed = true
+		existing.Spec.Template.Spec.Containers[0].VolumeMounts = desired.Spec.Template.Spec.Containers[0].VolumeMounts
+	}
+
+	return changed
+}
+
+func (r *ApicastReconciler) volumesMutator(desired, existing *appsv1.DeploymentConfig) bool {
+	changed := false
+
+	if !reflect.DeepEqual(existing.Spec.Template.Spec.Volumes, desired.Spec.Template.Spec.Volumes) {
+		changed = true
+		existing.Spec.Template.Spec.Volumes = desired.Spec.Template.Spec.Volumes
```

Comment:
Adding a note here: we need to be aware that this might have an impact on existing users installations. I have the feeling some users might be manually creating volumes for it, and adding this will revert those changes.
I am aware we've never told it's supported but still.

Reply:
Yes, that is why I underlined this very clearly in the description. IMO, this is something the operator can do and should do. The CRD should be the public interface to add volumes, like implemented in this PR for custom policies. 

Anyway, I do not have a strong position on this. If you think this can be a source of issues, I can remove the volume generic reconciliation. I would have to add specific reconciliation for the expected volumes in order to support adding and removing custom policies in the CR.

Reply:
I think this might led to issues. Although I agree that is something the operator can and should do. I don't have a strong position to this either. I was just adding a note so we are aware that if we go ahead with this we might receive some feedback about it

Reply:
Implemented specific reconciliation only for volumes (and volumemounts) derived from the custom policy conf. It allows manually added volumes and will not remove them in the reconcilliation loop.